src/CommitmentsV2/SFA.DAS.CommitmentsV2/Services/TransferRequestDomainService.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.CommitmentsV2.Application.Queries.GetTransferRequest;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Types;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.CommitmentsV2.Services
+{
+    public class TransferRequestDomainService : ITransferRequestDomainService
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+        private readonly ILogger<TransferRequestDomainService> _logger;
+
+        public TransferRequestDomainService(Lazy<ProviderCommitmentsDbContext> dbContext,
+            ILogger<TransferRequestDomainService> logger)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+        }
+
+        public async Task ApproveTransferRequest(long transferRequestId, UserInfo userInfo, DateTime approvedOn, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var transferRequest = await _dbContext.Value.TransferRequests
+                    .Include(c => c.Cohort)
+                    .Where(x => x.Id == transferRequestId)
+                    .SingleAsync(cancellationToken);
+
+                if (transferRequest.Status == TransferApprovalStatus.Approved)
+                {
+                    _logger.LogWarning($"Transfer Request {transferRequest.Id} has already been approved");
+                    return;
+                }
+
+                transferRequest.Approve(userInfo, approvedOn);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Error processing {nameof(ApproveTransferRequest)}");
+                throw;
+            }
+        }
+
+        public async Task RejectTransferRequest(long transferRequestId, UserInfo userInfo, DateTime rejectedOn, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var transferRequest = await _dbContext.Value.TransferRequests
+                    .Include(c => c.Cohort)
+                    .Where(x => x.Id == transferRequestId)
+                    .SingleAsync();
```

Comment:
```suggestion
                var transferRequest = await _dbContext.Value.TransferRequests
                    .Include(c => c.Cohort)
                    .SingleAsync(x => x.Id == transferRequestId, cancellationToken);
```

Reply:
additionally this duplicated query could be moved to a private method.

Reply:
Done